docs/contracts.rst

```diff
@@ -1406,24 +1406,27 @@ Using For
 
 The directive ``using A for B;`` can be used to attach library
 functions (from the library ``A``) to any type (``B``).
-These functions will receive the object they are called on
-as their first parameter (like the ``self`` variable in
-Python).
 
 The effect of ``using A for *;`` is that the functions from
-the library ``A`` are attached to any type.
+the library ``A`` are attached to *any* type.
 
-In both situations, all functions, even those where the
-type of the first parameter does not match the type of
-the object, are attached. The type is checked at the
+By including a module, its data types including library functions are
+available without having to add further code.
+
+In both situations, *all* functions in the library are attached,
+even those where the type of the first parameter does not
+match the type of the object. The type is checked at the
 point the function is called and function overload
 resolution is performed.
 
-The ``using A for B;`` directive is active for the current
-scope, which is limited to a contract for now but will
-be lifted to the global scope later, so that by including
-a module, its data types including library functions are
-available without having to add further code.
+The ``using A for B;`` directive is active only within the current
+contract, including within all of its functions and has no effect
+outside of the contract in which it is use. The directive
+may only be used inside a contract, not inside any of its functions.
```

Comment:
I would really prefer if you could keep the changes to a minimum and perhaps only add this single sentence.

Reply:
OK, all text reverted aside from the addition and one or two minor formattings (para break) and changing 'module' to 'library' (which I shall alter to any other term you prefer if necessary).